README.md

```diff
@@ -208,36 +216,349 @@ normalization to JSON API.
 
 #### Caching
 
-TODO: talk about cachekey
+When `cacheKey` is provided the response payload is cached.  This can be useful
+to show, eg dashboard data or any other data that changes over time.
+
+Consider the following:
+
+```js
+store.queryURL('/newsfeed/latest', { cacheKey: 'newsfeed.latest', backgroundReload: true });
+```
+
+In this example, the first time the user visits a route that makes this query,
+the promise will wait to resolve until the request completes.  The second time
+the request is made the promise will resolve immediately with the cached values
+while loading fresh values in the background.
+
+Note that what is actually cached is the result: ie either a `MegamorphicModel`
+or, more likely, a `RecordArray` of `MegamorphicModel`s.  
+
+---
+
+It is possible to do the same thing in stock Ember Data by making a `DS.Model`
+class to wrap your search results and querying via:
+
+```js
+// app/models/news-feed.js
+import DS from 'ember-data';
+export DS.Model.extend({
+  feedItems: DS.hasMany('feed-item'),
+});
+
+// somewhere, presumably in a route
+store.findRecord('news-feed', 'latest', { backgroundReload: true });
+```
+
+As with ember-m3 generally, similar functionality is provided without the need
+to create models and relationships within your app code.
+
+##### Cache Eviction
+
+Because models (or `RecordArray`s of models are cached) the cache can be emptied
+automatically when the models are unloaded.  In the case of `RecordArray`s of
+models, the entire cache entry is evicted if *any* of the member models is
+unloaded.
+
 
 ## Schema
 
+You have to register a schema to tell ember-m3 what types it should be enabled
```

Comment:
this should include, what is a schema, and when it is appropriate to use one, and how they differ from models.